torch/lib/c10d/reducer.hpp

```diff
@@ -205,6 +205,10 @@ class Reducer {
       torch::autograd::Variable& variable,
       GradCallback&& cb);
 
+  void set_env_variables();
```

Comment:
Nit: would be good to move these into an anonymous namespace or a namespace such as `logging` and have it take in the ddp_logging_data by pointer instead, so we can separate business logic from logging code easily.

Reply:
sounds good, let me put it into anonymous namespaces

Reply:
had a second thought, it is very hard to separate reducer and logging codes actually, as logging codes are trying to log reducer's states. if we move out logging codes from reducer, we have to pass almost all of reducer's private members to logging functions. that is probably not what we want. so keeping the logging in reducer for now.